server/src/main/java/de/uftos/services/StudentGroupService.java

```diff
@@ -145,11 +145,10 @@ public StudentGroupResponseDto update(String id, StudentGroupRequestDto groupReq
 
     List<Grade> grades =
         this.gradeRepository.findAllById(groupRequest.gradeIds());
-    List<Grade> allGrades = this.gradeRepository.findAll();
-    for (Grade grade : allGrades) {
-      grade.getStudentGroups().remove(group);
-    }
     for (Grade grade : grades) {
+      if (grade.getStudentGroups().contains(group)) {
```

Comment:
What about the grades that are no longer present in the `groupRequest`, but where previously referencing the group. These now reference a grade that they shouldn't.

You could just get the old group saved in the db before replacing it and remove the group from all grades in  the difference of the new and old groups grades.

Why use an if with a continue and not just `!(grade.getStudentGroups().contains(group))` as the condition and the add and save in the body?

Reply:
You are right, I forgot to do that.

I personally prefer early returns, but in this case in doesn't make that big of a difference